eZ/Publish/Core/Limitation/LocationLimitationType.php

```diff
@@ -125,25 +128,33 @@ public function evaluate(APILimitationValue $value, APIUserReference $currentUse
         } elseif ($object instanceof VersionInfo) {
             $object = $object->getContentInfo();
         } elseif (!$object instanceof ContentInfo) {
-            throw new InvalidArgumentException(
-                '$object',
-                'Must be of type: ContentCreateStruct, Content, VersionInfo or ContentInfo'
-            );
+            // As this is Role limitation we need to signal abstain on unsupported $object
+            return self::ACCESS_ABSTAIN;
         }
 
+        $targets = array_filter($targets, function ($target) {
+            return !$target instanceof Version;
+        });
+
         // Load locations if no specific placement was provided
-        if ($targets === null) {
-            if ($object->published) {
+        if (empty($targets)) {
+            // Skip check if content is in trash and no location is provided to check against
+            if ($object->isTrashed()) {
+                return self::ACCESS_ABSTAIN;
+            }
+
+            if ($object->isPublished()) {
                 $targets = $this->persistence->locationHandler()->loadLocationsByContent($object->id);
             } else {
-                // @todo Need support for draft locations to to work correctly
+                // @todo Need support for draft locations to work correctly
                 $targets = $this->persistence->locationHandler()->loadParentLocationsForDraftContent($object->id);
             }
         }
 
         foreach ($targets as $target) {
             if (!$target instanceof Location && !$target instanceof SPILocation) {
-                throw new InvalidArgumentException('$targets', 'Must contain objects of type: Location');
+                // As this is Role limitation we need to signal abstain on unsupported $targets
```

Comment:
The same here. LocationLimitation is not available when you assign a Role to the User or User Group. We get rid off `Versions` from `$targets`. I think that this change is not needed or at least explanation should be different.